examples/compiled/label_multi_line.vg.json

```diff
@@ -0,0 +1,173 @@
+{
+  "$schema": "https://vega.github.io/schema/vega/v5.json",
+  "background": "white",
+  "padding": 5,
+  "width": 200,
+  "height": 200,
+  "style": "cell",
+  "data": [
+    {
+      "name": "source_0",
+      "url": "data/stocks.csv",
+      "format": {"type": "csv", "parse": {"date": "date"}},
+      "transform": [
+        {
+          "field": "date",
+          "type": "timeunit",
+          "units": ["year"],
+          "as": ["year_date", "year_date_end"]
+        },
+        {
+          "type": "aggregate",
+          "groupby": ["year_date", "symbol"],
+          "ops": ["mean"],
+          "fields": ["price"],
+          "as": ["mean_price"]
+        }
+      ]
+    }
+  ],
+  "marks": [
+    {
+      "name": "pathgroup",
+      "type": "group",
+      "from": {
+        "facet": {
+          "name": "faceted_path_main",
+          "data": "source_0",
+          "groupby": ["symbol"]
+        }
+      },
+      "encode": {
+        "update": {
+          "width": {"field": {"group": "width"}},
+          "height": {"field": {"group": "height"}}
+        }
+      },
+      "marks": [
+        {
+          "name": "marks",
+          "type": "line",
+          "style": ["line"],
+          "sort": {"field": "datum[\"year_date\"]"},
+          "from": {"data": "faceted_path_main"},
+          "encode": {
+            "update": {
+              "stroke": {"scale": "color", "field": "symbol"},
+              "description": {
+                "signal": "\"date (year): \" + (timeFormat(datum[\"year_date\"], timeUnitSpecifier([\"year\"], {\"year-month\":\"%b %Y \",\"year-month-date\":\"%b %d, %Y \"}))) + \"; Mean of price: \" + (format(datum[\"mean_price\"], \"\")) + \"; symbol: \" + (isValid(datum[\"symbol\"]) ? datum[\"symbol\"] : \"\"+datum[\"symbol\"])"
+              },
+              "x": {"scale": "x", "field": "year_date"},
+              "y": {"scale": "y", "field": "mean_price"},
+              "defined": {
+                "signal": "isValid(datum[\"year_date\"]) && isFinite(+datum[\"year_date\"]) && isValid(datum[\"mean_price\"]) && isFinite(+datum[\"mean_price\"])"
+              }
+            }
+          }
+        }
+      ]
+    },
+    {
+      "name": "marks_label",
+      "type": "text",
+      "style": ["text"],
+      "sort": {"field": "datum[\"year_date\"]"},
+      "from": {"data": "pathgroup"},
+      "encode": {
+        "update": {
+          "fill": {"value": "black"},
```

Comment:
The tricky bit is that we shouldn't apply color to area's label, nor to stacked bar's labels.

Reply:
Could we have an option in the label encoding channel `detail: true | false`.
Then, the label would encode all the detail from the base mark that encode the same field.

Reply:
Or, this might work:
```js
{
  ...
  "encode": {
    "color": {"field": "a", "type": "nominal"},
    "label": {
      "field": "a",
      // or
      "from_detail": "color",  // need a better name

      ...other configs
    }
  }
}
```

Reply:
That's one approach, and we need a good name for it: 
```
"label": {
  "field": ...,
  "apply/encode": "color" // similar to your from_detail above, but 
}
```

However, I think I like this format more: 

```
"label": {
  "field": ...,
  "color": ... where color can be `null` (use the mark's color) or `string` (color name/hex`)
}
```

Reply:
I see. Can it be like this:
```js
"label": {
  "field": ...,
  "color": "inherit" | {value: "#fffff"}  
}
```

Reply:
It can, but do we think that "inherit" should be used more often than custom color? (If custom color is more often, then this `{value: ...}` seems an overkill.)


By the way, maybe the term `inherit` isn't bad.  Plus, thinking more, we also have `mark` already, so maybe the first format you proposed is possibly better and `inherit` could be a good keyword for it

```
"label": {
  "field": ...,
  "inherit": "color" | ["color", "opacity"] // similar to your from_detail above
}
```

Reply:
@domoritz  opinions?

Reply:
We also have to think about (1) how encoding inherits each encoding channel and (2) what is the default encoding channel to inherit for each mark type.
For example:
(1) In a line chart, the color in the label should inherit the line's stroke color.
(2) As you mentioned, area should not inherit color by default, but line should.